engines/bops_preapps/app/controllers/bops_preapps/tasks_controller.rb

```diff
@@ -4,14 +4,24 @@ module BopsPreapps
   class TasksController < ApplicationController
     include BopsCore::TasksController
 
-    before_action :set_planning_application, only: %i[show]
-    before_action :build_form, only: %i[edit update]
+    before_action :authenticate_user!
```

Comment:
Should this sit in an `authentication_controller`?
Then we can inherit from there.
I Imagine the tasks controller won't be the only one we do

Reply:
I wasn't sure it was worth it when there's currently only the one controller but if we're pretty sure we're going to have non-task sections within the Preapp engine then perhaps it makes sense.

Reply:
Actually, yeah, we already do have more than one controller in this engine.